docs/madr/decisions/056-reachable-svc-mesh-service.md

```diff
@@ -0,0 +1,254 @@
+        // Namespace of the targeted object.
+        //  +optional
+        string namespace = 4;
+        // Labels of the targeted object
+        map<string, string> labels = 5;
+      }
+```
+
+This enhancement should make it easier to match the reachable services provided by the user with the generated or provided outbounds. There is a disadventage that makes it maybe not the best option. If we expose it to the user, it might configure one `BackendRef` that reference many MeshServices. 
+
+Instead we can retrive `MeshService` and match based on its dpTags.
+
+### Autoreachable services
+
+From a code perspective, we used to build a map of supported tags: `kuma.io/service`, `k8s.kuma.io/namespace`, `k8s.kuma.io/service-name`, and `k8s.kuma.io/service-port`, and later match `MeshTrafficPermission` to dataplanes with these tags. Since the introduction of `MeshService` and `MeshExternalService`, we might not need to iterate over all dataplanes. Instead, we can match all `MeshService` and `MeshExternalService` entries. Based on this information, we know which dataplanes need to be configured. The only limitation is that `MeshService` doesn't carry all tags that a dataplane might have, which means that when using `MeshSubset`, users should be allowed to use the same tags as `dpTags` defined in `MeshService`.
+
+Algorithm:
+
+1. Take each`MeshService`
+2. For each `MeshService`, take tags from dpTags (`+kuma.io/zone`, +`kuma.io/origin`) and build a fake "DPP" with one inbound
+3. Execute MatchedPolicies on this "DPP" with a list of `MeshTrafficPermission` to get rules, in this case we have to trim tags from `MeshTrafficPermission` to have only one from `DPP`.
+4. Store rules for each `MeshService`.
+5. When we build XDS config for client DPP and check if we should add a cluster/listener for given `MeshService`, execute rules for `MeshService` using client DPP.
+
+Targetting:
+
+When a user wants to create a `MeshTrafficPermission` that targets a `MeshService`, they cannot do it directly in the top-level TargetRef. Instead, they can use `MeshSubset` to select a specific `MeshService`
+
+```yaml
+kind: MeshService
+spec:
+  selector:
+    dpTags:
+      app: backend
+      k8s.kuma.io/namespace: xyz
+---
+kind: MeshTrafficPermission
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      app: backend
+```
+
+In the case of `MeshExternalService`, we need to decide how we want to target them (this needs to be clarified in a separate MADR). For now, we are going to provide all `MeshExternalService` entries until we figure out targeting.
+
+Option 1
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: mydomain
+  namespace: kuma-system
+  labels:
+    my.label.io/service: my-service 
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+    kuma.io/origin: zone
+spec:
+  destinations:
+  - address: 192.168.0.1
+    port: 9090 
+---
+kind: MeshTrafficPermission
+spec:
+  targetRef:
+    kind: MeshExternalService
+    name: mydomain-HASH
```

Comment:
why would we use a name with the hash?

Reply:
If the MeshExternalService is synced from global we need to use a resource name in the zone, if we create it on zone. I might write this example better